version: '3.3'
services:
  db:
    image: postgis/postgis:14-3.3
    restart: unless-stopped
    env_file: .env
    ports:
      - '${POSTGRES_LOCAL_PORT}:${POSTGRES_DOCKER_PORT}'
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      timeout: 20s
      retries: 10
    volumes:
      - ./src/main/resources/buildings.sql:/docker-entrypoint-initdb.d/buildings.sql
      - ./src/main/resources/roads.sql:/docker-entrypoint-initdb.d/roads.sql
      - ./src/main/resources/rivers.sql:/docker-entrypoint-initdb.d/rivers.sql
      - ./src/main/resources/init.sql:/docker-entrypoint-initdb.d/init.sql
      - ./src/main/resources/testData.sql:/docker-entrypoint-initdb.d/testData.sql
#  ai:
#    build: ../ADoTO-AI
#    container_name: prediction_module
#    env_file: .env
#    links:
#      - app
  app:
    depends_on:
      db:
        condition: service_healthy
    build: .
    env_file: .env
    restart: unless-stopped
    ports:
      - '${SPRING_LOCAL_PORT}:${SPRING_DOCKER_PORT}'
    volumes:
      - .m2:/root/.m2
volumes:
  db: